gems/quilt_rails/test/controllers/quilt/ui_controller_test.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Quilt
+  class UiControllerTest < ActionDispatch::IntegrationTest
+    include ActiveSupport::Testing::Isolation
+
+    def run(*)
+      with_integration_tests_allowed { super }
```

Comment:
I don't think this is an elegant solution. Best would be to boot the app in production/development mode (whatever you prefer). You can achieve this by exporting the `RAILS_ENV` variable before doing the the call to `require 'environment'` https://github.com/Shopify/quilt/pull/1566/files#diff-af0adb8e661fc01cefcfafc3ee8efe4eR35

Reply:
🤦 This doesn't work because the dummy app is booted in the test helper. Let me try and sort this out...

Reply:
Done. I had to keep this wrapped on `run` because I needed `Rails.env` changed only for this test case, it is used in others. Changing `RAILS_ENV` doesn't work because `Rails.env` gets memoized in other tests.

Reply:
> Changing RAILS_ENV doesn't work because Rails.env gets memoized in other tests.

Not sure I understand. Why do other tests matter ? Does changing `RAILS_ENV` breaks other tests ?

Reply:
https://github.com/rails/rails/blob/c6f02cbb56b6291533a3756bb5016964c76f5c20/railties/lib/rails.rb#L73

The problem is memoization, changing `RAILS_ENV` does nothing once it is memoized.

Reply:
Actually we can unset it in boot method, that will probably fix it if the fork _just dies_ as you say.